### PR TITLE
feat: export metrics

### PR DESCRIPTION
**Pull request by zac-li**
- Some changes are around isort fix
- Added pre-commit for isort and black
- Also added tests to assert metrics are correctly emitted/collected

### CODE REVIEW

File: lcserve/backend/gateway.py [hunk 1]
```diff
@@ -5,18 +5,21 @@
...
 from docarray import Document, DocumentArray
 from jina import Gateway
-from jina.logging.logger import JinaLogger
 from jina.enums import ProtocolType as GatewayProtocolType
+from jina.logging.logger import JinaLogger
 from jina.serve.runtimes.gateway.composite import CompositeGateway
 from jina.serve.runtimes.gateway.http.fastapi import FastAPIBaseGateway
-from pydantic import Field, ValidationError, create_model, BaseModel
+from opentelemetry.instrumentation.fastapi import FastAPIInstrumentor
+from opentelemetry.sdk.metrics import Counter
```
Comment (deepankarm) [inline lcserve/backend/gateway.py:21]: Please move these imports inside the functions that use them

**Review by deepankarm — COMMENTED** [review]
:+1: How can we add integration tests for this?

File: lcserve/flow.py [hunk 2]
```diff
@@ -351,6 +351,8 @@ def get_global_jcloud_args(app_id: str = None, name: str = APP_NAME) -> Dict:
                 },
                 'metrics': {
                     'enable': True,
+                    'host': 'http://opentelemetry-collector.monitor.svc.cluster.local',
```
Comment (deepankarm) [inline lcserve/flow.py:354]: Why is this needed on the client side?

Comment (zac-li) [reply lcserve/flow.py:354]: because by default `metrics` isn't enabled and would require host/port according to @tarrantro

Comment (tarrantro) [reply lcserve/flow.py:354]: That's because we user prometheus for metrics, otlp metrics is not stable as @winstonww mentioned. So we disable it to avoid duplicate metrics.

**Review by deepankarm — APPROVED** [review]
